### DISCUSSION TITLE
Replace / improve "ideal"-image

### DISCUSSION THREAD

**Discussion by levino** [facebook/docusaurus#9253] [Ideas]
The current image implementation `plugin-ideal-image` looks not very ideal to me. The created html output is not using very powerful features like the `srcset` artribute for `img` tags. I think there might be potential in using `nextjs`s `Image` component also for docusaurus and dropping the currently used dependency to `react-ideal-image` (or better the fork of the fork of the fork of it...).

Important features the new component should have:
- Use `loading="lazy"`
- Use `srcset`
- Optionally start with a `srcset` which has just one blurhash and replace that when the image is in the viewport with the "real" `srcset`.
- Set `width` and `height` correctly based on the input image.

**Comment by slorber** [comment]
I am aware this legacy plugin is not ideal but it's not my top priority to improve it atm, considering users can create their own plugins in userland.

If you want to work on a better image plugin for Docusaurus, you are welcome, and we might consider to make it the official one if it's proven successful
